### PR TITLE
gui: Add fix for wxGUI Data import Python error

### LINKED ISSUE #648: [Bug] wxGUI Data import Python error
Hello, 
When I export DEM data to GRASS GIS software, I get the error 'if not self.add.IsChecked () or returncode! = 0: RuntimeError: wrapped C / C ++ object of type CheckBox has been deleted'.

To Reproduce
File>İmport raster data>Simplifed raster import with reprojection> Source İmport > Data import > Error

Screenshots
Ekran Alıntısı1
Ekran Alıntısı2

 Operating System: Windows 10 Pro unlicensed, 320 Gb Hdd; 8 Gb Ram
Grass Gıs version 7.8 

What is the cause?
Anybody suggest a solution? Thank You!

### PR DESCRIPTION
**Pull request by ww2406**
This fixes #648 by adding a command tracker queue.

Currently, a user can trigger an error if they manually close the dialog box before the commands. This is due to the command completion callbacks depending on state from the UI dialog which is not accessible if the dialog is destroyed.

To address this, I added a command tracking queue to keep track of the number of commands executing. This PR then modifies the close event handler so that If this queue is not empty, the close command is interrupted and does not execute.

One opportunity might be to consolidate the command dispatch mechanism into one function so future maintainers do not need to explicitly invoke the proposed `_addToCommandQueue` function, but I wasn't sure if that was optimizing more than needed at the moment.

Look forward to any feedback you might have!

### CODE REVIEW

**Comment by echoix** [comment]
Great to see some progress on an older issue!

**Comment by petrasovaa** [comment]
@ww2406 thank you for looking into this.  I think this makes sense, I will leave just couple notes below.

File: gui/wxpython/modules/import_export.py [hunk 1]
```diff
@@ -122,16 +125,17 @@ def __init__(
         # buttons
         #
         # cancel
+        self._DEFAULT_CLOSE_BTN_TEXT = "Close dialog"
```
Comment (petrasovaa) [inline gui/wxpython/modules/import_export.py:128]: This should be translatable, here you are calling SetToolTip with the translated version and later with untranslated version.

File: gui/wxpython/modules/import_export.py [hunk 2]
```diff
@@ -270,6 +274,30 @@ def _validateOutputMapName(self):
                 return False
         return True
 
+    def _handleCloseEvent(self, *args, **kwargs):
+        if len(self._commands_running) > 0:
+            # prevent dialog close while async commands are running
+            return
```
Comment (petrasovaa) [inline gui/wxpython/modules/import_export.py:280]: This should probably use event.Veto:
https://docs.wxpython.org/wx.CloseEvent.html

File: gui/wxpython/modules/import_export.py [hunk 3]
```diff
@@ -270,6 +274,30 @@ def _validateOutputMapName(self):
...
+        self.Destroy()
+
+    def _addToCommandQueue(self):
+        self._commands_running.append(True)
+
+        # disable the dialog close button
+        self.btn_close.SetToolTip(_("Dialog cannot be closed while command is running."))
```
[the remark this answers is not in the document]
Comment (petrasovaa) [reply gui/wxpython/modules/import_export.py:287]: We are using Black for formatting: https://github.com/OSGeo/grass/blob/main/doc/development/style_guide.md

**Review by petrasovaa — APPROVED** [review]
Looks good to me, thank you for your contribution!

**Pull request merged**